### PR TITLE
fix(addon) upgrades ignored previous values

### LINKED ISSUE #3399: jx upgrade addon ignores any options used when the addon was installed
### Summary 

If you install an addon that uses some options in the helm chard (e.g. `jx create addon cloudbees`) then when you upgrade the addon with `jx upgrade` the options get reset to the defailts.

### Steps to reproduce the behaviour

install the cloudbees addon with sso enabled.
wait
upgrade the cloudbees addon

### Expected behavior

the cloudbees addon is installed and configured with SSO

### Actual behavior

the cloudbees addon is installed but is not configured to use SSO

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 [32m1.3.980[0m
jenkins x platform [32m0.0.3368[0m
Kubernetes cluster [32mv1.12.5-gke.5[0m
kubectl            [32mv1.13.0[0m
helm client        [32mv2.11.0+g2e55dbe[0m
helm server        [32mv2.11.0+g2e55dbe[0m
git                [32mgit version 2.16.2.windows.1[0m
Operating System   [32mWindows 10 Pro 1809 build 17763[0m
```

### Jenkins type

- [ ] Classic Jenkins
- [x] Serverless Jenkins

### PR DESCRIPTION
**Pull request by jtnord**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

when upgrading the addon any previous values for customosation where
lost.  Thic change tells helm to merge existing values with ones that
where used previously when upgrading.

#### Special notes for the reviewer(s)

Please think about any other usage that this may break, I hope I have confined it to just addons.

#### Which issue this PR fixes

fixes #3399

### CODE REVIEW

**Comment by jtnord** [comment]
/hold
do not auto merge this - need some careful eyes also want to run the tests in CI

File: pkg/jx/cmd/add_app_test.go [hunk 1]
```diff
@@ -188,20 +188,7 @@ func TestAddAppWithSecrets(t *testing.T) {
...
-		pegomock.AnyBool(),
-		pegomock.AnyInt(),
-		pegomock.AnyBool(),
-		pegomock.AnyBool(),
-		pegomock.AnyStringSlice(),
-		pegomock.AnyStringSlice(),
-		pegomock.EqString(kube.DefaultChartMuseumURL),
-		pegomock.AnyString(),
-		pegomock.AnyString())).
+	pegomock.When(testOptions.MockHelmer.UpgradeChart(pegomock.AnyString(), pegomock.EqString(name), pegomock.AnyString(), pegomock.EqString(version), pegomock.AnyBool(), pegomock.AnyInt(), pegomock.AnyBool(), pegomock.AnyBool(), pegomock.AnyStringSlice(), pegomock.AnyStringSlice(), pegomock.EqString(kube.DefaultChartMuseumURL), pegomock.AnyString(), pegomock.AnyString(), false)).
```
Comment (jtnord) [inline pkg/jx/cmd/add_app_test.go]: sorry for the fomatting, goland did it :)

Comment (pmuir) [reply pkg/jx/cmd/add_app_test.go]: You should be able to turn automatic wrapping of lines in Goland :-)

Comment (jtnord) [reply pkg/jx/cmd/add_app_test.go]: it is already set (hard wrap at 120), looks like the refactor engine ignores this. will fix them up manually whilst fixing the tests

**Pull request merged**